presto-hive/src/main/java/com/facebook/presto/hive/HiveMetadata.java

```diff
@@ -514,19 +515,20 @@ private PartitionStatistics getTableStatistics(SchemaTableName schemaTableName,
 
         ImmutableMap.Builder<String, PartitionStatistics> resultMap = ImmutableMap.builder();
 
-        Set<String> partitionNames = hivePartitions.stream().map(HivePartition::getPartitionId).collect(Collectors.toSet());
+        List<String> partitionNames = hivePartitions.stream().map(HivePartition::getPartitionId).collect(Collectors.toList());
         Map<String, Map<String, ColumnStatistics>> partitionColumnStatisticsMap =
-                metastore.getPartitionColumnStatistics(databaseName, tableName, partitionNames, tableColumns)
+                metastore.getPartitionColumnStatistics(databaseName, tableName, new HashSet<>(partitionNames), tableColumns)
                         .orElse(ImmutableMap.of());
 
-        for (HivePartition hivePartition : hivePartitions) {
-            String partitionId = hivePartition.getPartitionId();
-            Partition partition = metastore
-                    .getPartition(databaseName, tableName, toPartitionValues(partitionId))
-                    .orElseThrow(() -> new IllegalArgumentException(format("Could not get metadata for partition %s.%s.%s", databaseName, tableName, partitionId)));
-            Map<String, ColumnStatistics> partitionColumnStatistics = partitionColumnStatisticsMap.getOrDefault(partitionId, ImmutableMap.of());
-            resultMap.put(partitionId, readStatisticsFromParameters(partition.getParameters(), partitionColumnStatistics));
+        Map<String, Optional<Partition>> partitionsByNames = metastore.getPartitionsByNames(databaseName, tableName, partitionNames);
```

Comment:
How does this result in getting partition statistics in batches? It seems to be doing all partitions at once.